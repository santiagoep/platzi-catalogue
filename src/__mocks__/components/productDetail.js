export default {
  name: 'name',
  price: 150,
  description: 'description',
  images: [
    { src: '/icons/isotype.png', alt: 'Isotype' },
    { src: '/icons/isotype.png', alt: 'Isotype' }
  ],
  category: 'category'
};
