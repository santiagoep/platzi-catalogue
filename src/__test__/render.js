/* istanbul ignore file */
/* eslint-disable import/no-extraneous-dependencies */
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { makeMockedStore } from '@store/index';
import mockedStore from '@__mocks__/store/index';
import lightTheme from '@assets/styles/themes/light';

const WrapperComponent = ({ children, config, initialState }) => {
  if (config && config.theme && config.redux) {
    const store = makeMockedStore(initialState || mockedStore);
    return (
      <Provider store={store}>
        <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
      </Provider>
    );
  }
  if (config && config.redux) {
    const store = makeMockedStore(initialState || mockedStore);
    return <Provider store={store}>{children}</Provider>;
  }
  if (config && config.theme) {
    return <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>;
  }
  return children;
};

const renderWithProviders = (
  { redux, theme } = {},
  initialState,
  getMockedStore
) => {
  const renderProviders = (Component) => (
    <WrapperComponent
      config={{ redux, theme }}
      initialState={initialState}
      getMockedStore={getMockedStore}
    >
      {Component}
    </WrapperComponent>
  );

  return {
    mount: (Component) => mount(renderProviders(Component)),
    render: (Component) => render(renderProviders(Component)),
    shallow: (Component) => shallow(renderProviders(Component)),
    renderer: (Component) => Renderer.create(renderProviders(Component))
  };
};

export default renderWithProviders;
