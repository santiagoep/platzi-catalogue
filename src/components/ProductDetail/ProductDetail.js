import PropTypes from 'prop-types';

import Link from '@components/Link/Link';
import formatCurrency from '@utils/numbers';
import { Col, Row, Grid } from '@components/Grid/Grid';
import { StyledContainer } from './ProductDetail.styled';
import Typography from '@components/Typography/Typography';
import FakeButton from '@components/FakeButton/FakeButton';
import ProductDetailImages from '@components/ProductDetailImages/ProductDetailImages';

const ProductDetail = ({
  name,
  price,
  description,
  images,
  category,
  onBack
}) => (
  <Grid>
    <Row>
      <Col smallMobile='100%' tablet='50%'>
        <ProductDetailImages images={images} />
      </Col>
      <Col smallMobile='100%' tablet='50%'>
        <StyledContainer>
          <Typography
            className='product-detail__category-name'
            type='paragraph4'
            color='quaternary'
            fontWeight={400}
            as='p'
          >
            {category?.name}
          </Typography>
          <Typography
            className='product-detail__name'
            type='subtitle1'
            color='secondary'
            fontWeight='bold'
            as='h2'
          >
            {name}
          </Typography>
          <Typography
            className='product-detail__price'
            type='paragraph3'
            color='primary'
            fontWeight='bold'
            as='p'
          >
            {formatCurrency(price)}
          </Typography>
          <Typography
            className='product-detail__description'
            type='paragraph1'
            color='primary'
            as='p'
          >
            {description}
          </Typography>
          <Link href='/' onClick={onBack}>
            <FakeButton>
              <Typography color='tertiary' fontWeight='bold'>
                Volver
              </Typography>
            </FakeButton>
          </Link>
        </StyledContainer>
      </Col>
    </Row>
  </Grid>
);

ProductDetail.propTypes = {
  name: PropTypes.string,
  price: PropTypes.number,
  images: PropTypes.arrayOf(
    PropTypes.shape({
      src: PropTypes.string,
      alt: PropTypes.string
    })
  ),
  description: PropTypes.string,
  category: PropTypes.shape({ name: PropTypes.string }),
  onBack: PropTypes.func
};

ProductDetail.defaultProps = {
  name: '',
  price: undefined,
  images: [],
  description: '',
  category: {},
  onBack: () => {}
};

export default ProductDetail;
