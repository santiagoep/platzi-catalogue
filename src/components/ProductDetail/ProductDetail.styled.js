import styled from 'styled-components';

export default styled.div`
  margin: ${({ theme }) => theme.utils.spacing(5)};
  padding: ${({ theme }) => theme.utils.spacing(5)};
  box-shadow: ${({ theme }) => theme.boxShadows.main};
  border-radius: ${({ theme }) => theme.borderRadius.medium};
  background-color: ${({ theme }) => theme.colors.bg.secondary};
  > * {
    text-align: center;
    ${({ theme }) => theme.breakpoints.tablet} {
      text-align: left;
    }
  }
  .product-detail__description {
    margin: ${({ theme }) => theme.utils.spacing(5, 2)};
  }
  .product-detail__category-name {
    margin-bottom: ${({ theme }) => theme.utils.spacing(3)};
  }
`;
