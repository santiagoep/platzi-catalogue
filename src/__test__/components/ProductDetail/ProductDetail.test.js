import render from '../../render';
import productDetail from '@__mocks__/components/productDetail';
import ProductDetail from '@components/ProductDetail/ProductDetail';

describe('<ProductDetail />', () => {
  it('Should render snapshot', () => {
    const { mount } = render({ theme: true });
    const component = mount(<ProductDetail {...productDetail} />);
    expect(component).toMatchSnapshot();
  });
  it('Should on back', () => {
    const { mount } = render({ theme: true });
    const spy = jest.fn();
    const component = mount(<ProductDetail onBack={spy} {...productDetail} />);
    component.find('a').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
