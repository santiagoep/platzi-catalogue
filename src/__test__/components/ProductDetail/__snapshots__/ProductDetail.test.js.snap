// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetail /> Should render snapshot 1`] = `
<WrapperComponent
  config={
    Object {
      "redux": undefined,
      "theme": true,
    }
  }
>
  <Component
    theme={
      Object {
        "borderRadius": Object {
          "button": "100px",
          "large": "18px",
          "medium": "8px",
          "none": "unset",
          "small": "3px",
          "total": "50%",
        },
        "boxShadows": Object {
          "main": "1px 2px 4px 0 rgba(21,99,157,.16)",
          "secondary": "0 0 0 1px #0791e6",
        },
        "breakpoints": Object {
          "desktop": "@media only screen and (min-width: 1280px)",
          "laptop": "@media only screen and (min-width: 1024px)",
          "mobile": "@media only screen and (min-width: 320px)",
          "screen": "@media only screen and (min-width: 1920px)",
          "smallMobile": "@media only screen and (min-width: 0px)",
          "tablet": "@media only screen and (min-width: 640px)",
        },
        "colors": Object {
          "bg": Object {
            "primary": "linear-gradient(90deg,#1c3643,#273b47 25%,#1e5372)",
            "quaternary": "#F3F8FB",
            "secondary": "white",
            "tertiary": "#0791e6",
          },
          "border": Object {
            "primary": "#A8B1C0",
            "secondary": "white",
          },
          "font": Object {
            "primary": "#131D30",
            "quaternary": "#A8B1C0",
            "secondary": "#98ca3f",
            "tertiary": "white",
          },
          "separator": Object {
            "primary": "#A8B1C0",
            "secondary": "#0791e6",
          },
        },
        "containerWidth": Object {
          "desktop": Object {
            "maxWidth": "85%",
            "width": "85%",
          },
          "fluid": Object {
            "maxWidth": "100%",
            "width": "100%",
          },
          "limitFluid": Object {
            "maxWidth": "1920px",
            "width": "100%",
          },
          "mobile": Object {
            "maxWidth": "92%",
            "width": "92%",
          },
          "screen": Object {
            "maxWidth": "75%",
            "width": "75%",
          },
          "smallMobile": Object {
            "maxWidth": "92%",
            "width": "92%",
          },
          "tablet": Object {
            "maxWidth": "85%",
            "width": "85%",
          },
        },
        "font": Object {
          "family": Object {
            "primary": "'Lato', sans-serif",
          },
          "letterSpacing": Object {
            "medium": "0.5rem",
            "normal": "0rem",
          },
          "lineHeight": Object {
            "medium": "0.6rem",
            "normal": "normal",
          },
          "size": Object {
            "h1": "3.75rem",
            "h2": "3rem",
            "h3": "2.5rem",
            "h4": "2.25rem",
            "h5": "1.625rem",
            "h6": "1.5rem",
            "help": "0.625em",
            "paragraph1": "1.125rem",
            "paragraph2": "1rem",
            "paragraph3": "0.875rem",
            "paragraph4": "0.75rem",
            "small": "0.5rem",
            "subtitle1": "1.1375rem",
            "subtitle2": "1.25rem",
            "subtitle3": "1.375rem",
          },
        },
        "name": "LIGHT",
        "transitions": Object {
          "button": "background-color 0.4s ease, color 0.4s ease, border-color 0.4s ease",
        },
        "utils": Object {
          "spacing": [Function],
        },
        "zIndex": Object {
          "backward": -1,
          "fixed": 2,
          "forward": 1,
          "higher": 3,
          "initial": 0,
          "super": 100,
        },
      }
    }
  >
    <ProductDetail
      category="category"
      description="description"
      images={
        Array [
          Object {
            "alt": "Isotype",
            "src": "/icons/isotype.png",
          },
          Object {
            "alt": "Isotype",
            "src": "/icons/isotype.png",
          },
        ]
      }
      name="name"
      onBack={[Function]}
      price={150}
    >
      <Container>
        <Containerstyled>
          <div
            className="Containerstyled-qmu1bg-0 fwZIDA"
          >
            <Row>
              <Rowstyled>
                <div
                  className="Rowstyled-sc-76dq7o-0 kjMbCD"
                >
                  <Col
                    smallMobile="100%"
                    tablet="50%"
                  >
                    <Colstyled
                      smallMobile="100%"
                      tablet="50%"
                    >
                      <div
                        className="Colstyled-k6pt3-0 dePlOp"
                      >
                        <ProductDetailImages
                          data-testid="product-detail__images"
                          images={
                            Array [
                              Object {
                                "alt": "Isotype",
                                "src": "/icons/isotype.png",
                              },
                              Object {
                                "alt": "Isotype",
                                "src": "/icons/isotype.png",
                              },
                            ]
                          }
                        >
                          <Slider
                            autoplay={true}
                            autoplaySpeed={3000}
                            data-testid="product-detail__images"
                            dots={true}
                            fade={true}
                            infinite={true}
                            lazyLoad={true}
                            pauseOnHover={true}
                            slidesToScroll={1}
                            slidesToShow={1}
                            speed={1000}
                          >
                            <Sliderstyled__StyledContainer>
                              <div
                                className="Sliderstyled__StyledContainer-sc-1d9pqbc-0 dGMmUB"
                              >
                                <Component
                                  appendDots={[Function]}
                                  autoplay={true}
                                  autoplaySpeed={3000}
                                  data-testid="product-detail__images"
                                  dots={true}
                                  fade={true}
                                  infinite={true}
                                  lazyLoad={true}
                                  nextArrow={<Sliderstyled__StyledNextArrow />}
                                  pauseOnHover={true}
                                  prevArrow={<Sliderstyled__StyledPrevArrow />}
                                  slidesToScroll={1}
                                  slidesToShow={1}
                                  speed={1000}
                                >
                                  <div
                                    appendDots={[Function]}
                                    autoplay={true}
                                    autoplaySpeed={3000}
                                    data-testid="product-detail__images"
                                    dots={true}
                                    fade={true}
                                    infinite={true}
                                    lazyLoad={true}
                                    nextArrow={<Sliderstyled__StyledNextArrow />}
                                    pauseOnHover={true}
                                    prevArrow={<Sliderstyled__StyledPrevArrow />}
                                    slidesToScroll={1}
                                    slidesToShow={1}
                                    speed={1000}
                                  >
                                    <Slide
                                      key="image-slide-0"
                                    >
                                      <ProductDetailImage
                                        alt="Isotype"
                                        src="/icons/isotype.png"
                                      >
                                        <ProductDetailImagestyled>
                                          <div
                                            className="ProductDetailImagestyled-qsmc9q-0 jUkCNe"
                                          >
                                            <Image
                                              alt="Isotype"
                                              layout="fill"
                                              loading="eager"
                                              objectFit="cover"
                                              priority={true}
                                              quality={100}
                                              src="/icons/isotype.png"
                                            >
                                              <div
                                                style={
                                                  Object {
                                                    "bottom": 0,
                                                    "boxSizing": "border-box",
                                                    "display": "block",
                                                    "left": 0,
                                                    "margin": 0,
                                                    "overflow": "hidden",
                                                    "position": "absolute",
                                                    "right": 0,
                                                    "top": 0,
                                                  }
                                                }
                                              >
                                                <img
                                                  alt="Isotype"
                                                  decoding="async"
                                                  sizes="(max-width: 640px) 640px, (max-width: 750px) 750px, (max-width: 828px) 828px, (max-width: 1080px) 1080px, (max-width: 1200px) 1200px, (max-width: 1920px) 1920px, (max-width: 2048px) 2048px, 3840px"
                                                  src="/_next/image?url=%2Ficons%2Fisotype.png&w=3840&q=100"
                                                  srcSet="/_next/image?url=%2Ficons%2Fisotype.png&w=640&q=100 640w, /_next/image?url=%2Ficons%2Fisotype.png&w=750&q=100 750w, /_next/image?url=%2Ficons%2Fisotype.png&w=828&q=100 828w, /_next/image?url=%2Ficons%2Fisotype.png&w=1080&q=100 1080w, /_next/image?url=%2Ficons%2Fisotype.png&w=1200&q=100 1200w, /_next/image?url=%2Ficons%2Fisotype.png&w=1920&q=100 1920w, /_next/image?url=%2Ficons%2Fisotype.png&w=2048&q=100 2048w, /_next/image?url=%2Ficons%2Fisotype.png&w=3840&q=100 3840w"
                                                  style={
                                                    Object {
                                                      "border": "none",
                                                      "bottom": 0,
                                                      "boxSizing": "border-box",
                                                      "display": "block",
                                                      "height": 0,
                                                      "left": 0,
                                                      "margin": "auto",
                                                      "maxHeight": "100%",
                                                      "maxWidth": "100%",
                                                      "minHeight": "100%",
                                                      "minWidth": "100%",
                                                      "objectFit": "cover",
                                                      "objectPosition": undefined,
                                                      "padding": 0,
                                                      "position": "absolute",
                                                      "right": 0,
                                                      "top": 0,
                                                      "visibility": "visible",
                                                      "width": 0,
                                                    }
                                                  }
                                                />
                                              </div>
                                            </Image>
                                          </div>
                                        </ProductDetailImagestyled>
                                      </ProductDetailImage>
                                    </Slide>
                                    <Slide
                                      key="image-slide-1"
                                    >
                                      <ProductDetailImage
                                        alt="Isotype"
                                        src="/icons/isotype.png"
                                      >
                                        <ProductDetailImagestyled>
                                          <div
                                            className="ProductDetailImagestyled-qsmc9q-0 jUkCNe"
                                          >
                                            <Image
                                              alt="Isotype"
                                              layout="fill"
                                              loading="eager"
                                              objectFit="cover"
                                              priority={true}
                                              quality={100}
                                              src="/icons/isotype.png"
                                            >
                                              <div
                                                style={
                                                  Object {
                                                    "bottom": 0,
                                                    "boxSizing": "border-box",
                                                    "display": "block",
                                                    "left": 0,
                                                    "margin": 0,
                                                    "overflow": "hidden",
                                                    "position": "absolute",
                                                    "right": 0,
                                                    "top": 0,
                                                  }
                                                }
                                              >
                                                <img
                                                  alt="Isotype"
                                                  decoding="async"
                                                  sizes="(max-width: 640px) 640px, (max-width: 750px) 750px, (max-width: 828px) 828px, (max-width: 1080px) 1080px, (max-width: 1200px) 1200px, (max-width: 1920px) 1920px, (max-width: 2048px) 2048px, 3840px"
                                                  src="/_next/image?url=%2Ficons%2Fisotype.png&w=3840&q=100"
                                                  srcSet="/_next/image?url=%2Ficons%2Fisotype.png&w=640&q=100 640w, /_next/image?url=%2Ficons%2Fisotype.png&w=750&q=100 750w, /_next/image?url=%2Ficons%2Fisotype.png&w=828&q=100 828w, /_next/image?url=%2Ficons%2Fisotype.png&w=1080&q=100 1080w, /_next/image?url=%2Ficons%2Fisotype.png&w=1200&q=100 1200w, /_next/image?url=%2Ficons%2Fisotype.png&w=1920&q=100 1920w, /_next/image?url=%2Ficons%2Fisotype.png&w=2048&q=100 2048w, /_next/image?url=%2Ficons%2Fisotype.png&w=3840&q=100 3840w"
                                                  style={
                                                    Object {
                                                      "border": "none",
                                                      "bottom": 0,
                                                      "boxSizing": "border-box",
                                                      "display": "block",
                                                      "height": 0,
                                                      "left": 0,
                                                      "margin": "auto",
                                                      "maxHeight": "100%",
                                                      "maxWidth": "100%",
                                                      "minHeight": "100%",
                                                      "minWidth": "100%",
                                                      "objectFit": "cover",
                                                      "objectPosition": undefined,
                                                      "padding": 0,
                                                      "position": "absolute",
                                                      "right": 0,
                                                      "top": 0,
                                                      "visibility": "visible",
                                                      "width": 0,
                                                    }
                                                  }
                                                />
                                              </div>
                                            </Image>
                                          </div>
                                        </ProductDetailImagestyled>
                                      </ProductDetailImage>
                                    </Slide>
                                  </div>
                                </Component>
                              </div>
                            </Sliderstyled__StyledContainer>
                          </Slider>
                        </ProductDetailImages>
                      </div>
                    </Colstyled>
                  </Col>
                  <Col
                    smallMobile="100%"
                    tablet="50%"
                  >
                    <Colstyled
                      smallMobile="100%"
                      tablet="50%"
                    >
                      <div
                        className="Colstyled-k6pt3-0 dePlOp"
                      >
                        <ProductDetailstyled>
                          <div
                            className="ProductDetailstyled-sc-9inifs-0 jGytkp"
                          >
                            <Typography
                              as="p"
                              className="product-detail__category-name"
                              color="quaternary"
                              data-testid="product-detail__category-name"
                              fontFamily="primary"
                              fontStyle="normal"
                              fontWeight={400}
                              letterSpacing="normal"
                              lineHeight="normal"
                              textDecoration="none"
                              textTransform="none"
                              type="paragraph4"
                            >
                              <Typographystyled
                                as="p"
                                className="product-detail__category-name"
                                color="quaternary"
                                data-testid="product-detail__category-name"
                                fontFamily="primary"
                                fontStyle="normal"
                                fontWeight={400}
                                letterSpacing="normal"
                                lineHeight="normal"
                                textDecoration="none"
                                textTransform="none"
                                type="paragraph4"
                              >
                                <p
                                  className="Typographystyled-sc-1pbtdtz-0 hxSQkE product-detail__category-name"
                                  color="quaternary"
                                  data-testid="product-detail__category-name"
                                  fontFamily="primary"
                                  fontStyle="normal"
                                  fontWeight={400}
                                  letterSpacing="normal"
                                  textDecoration="none"
                                  type="paragraph4"
                                />
                              </Typographystyled>
                            </Typography>
                            <Typography
                              as="h2"
                              className="product-detail__name"
                              color="secondary"
                              data-testid="product-detail__name"
                              fontFamily="primary"
                              fontStyle="normal"
                              fontWeight="bold"
                              letterSpacing="normal"
                              lineHeight="normal"
                              textDecoration="none"
                              textTransform="none"
                              type="subtitle1"
                            >
                              <Typographystyled
                                as="h2"
                                className="product-detail__name"
                                color="secondary"
                                data-testid="product-detail__name"
                                fontFamily="primary"
                                fontStyle="normal"
                                fontWeight="bold"
                                letterSpacing="normal"
                                lineHeight="normal"
                                textDecoration="none"
                                textTransform="none"
                                type="subtitle1"
                              >
                                <h2
                                  className="Typographystyled-sc-1pbtdtz-0 Falna product-detail__name"
                                  color="secondary"
                                  data-testid="product-detail__name"
                                  fontFamily="primary"
                                  fontStyle="normal"
                                  fontWeight="bold"
                                  letterSpacing="normal"
                                  textDecoration="none"
                                  type="subtitle1"
                                >
                                  name
                                </h2>
                              </Typographystyled>
                            </Typography>
                            <Typography
                              as="p"
                              className="product-detail__price"
                              color="primary"
                              data-testid="product-detail__price"
                              fontFamily="primary"
                              fontStyle="normal"
                              fontWeight="bold"
                              letterSpacing="normal"
                              lineHeight="normal"
                              textDecoration="none"
                              textTransform="none"
                              type="paragraph3"
                            >
                              <Typographystyled
                                as="p"
                                className="product-detail__price"
                                color="primary"
                                data-testid="product-detail__price"
                                fontFamily="primary"
                                fontStyle="normal"
                                fontWeight="bold"
                                letterSpacing="normal"
                                lineHeight="normal"
                                textDecoration="none"
                                textTransform="none"
                                type="paragraph3"
                              >
                                <p
                                  className="Typographystyled-sc-1pbtdtz-0 ecdERZ product-detail__price"
                                  color="primary"
                                  data-testid="product-detail__price"
                                  fontFamily="primary"
                                  fontStyle="normal"
                                  fontWeight="bold"
                                  letterSpacing="normal"
                                  textDecoration="none"
                                  type="paragraph3"
                                >
                                  $ 150
                                </p>
                              </Typographystyled>
                            </Typography>
                            <Typography
                              as="p"
                              className="product-detail__description"
                              color="primary"
                              data-testid="product-detail__description"
                              fontFamily="primary"
                              fontStyle="normal"
                              fontWeight="normal"
                              letterSpacing="normal"
                              lineHeight="normal"
                              textDecoration="none"
                              textTransform="none"
                              type="paragraph1"
                            >
                              <Typographystyled
                                as="p"
                                className="product-detail__description"
                                color="primary"
                                data-testid="product-detail__description"
                                fontFamily="primary"
                                fontStyle="normal"
                                fontWeight="normal"
                                letterSpacing="normal"
                                lineHeight="normal"
                                textDecoration="none"
                                textTransform="none"
                                type="paragraph1"
                              >
                                <p
                                  className="Typographystyled-sc-1pbtdtz-0 bzcDUm product-detail__description"
                                  color="primary"
                                  data-testid="product-detail__description"
                                  fontFamily="primary"
                                  fontStyle="normal"
                                  fontWeight="normal"
                                  letterSpacing="normal"
                                  textDecoration="none"
                                  type="paragraph1"
                                >
                                  description
                                </p>
                              </Typographystyled>
                            </Typography>
                            <InternalLink
                              className=""
                              data-testid="product-detail__go-back"
                              href="/"
                              onClick={[Function]}
                              target="_blank"
                            >
                              <Component
                                href="/"
                                onClick={[Function]}
                                passHref={false}
                              >
                                <div
                                  href="/"
                                  onClick={[Function]}
                                  passHref={false}
                                >
                                  <a
                                    className=""
                                    data-testid="product-detail__go-back"
                                  >
                                    <FakeButton>
                                      <Fakebuttonstyled>
                                        <div
                                          className="Fakebuttonstyled-gl3kvr-0 dIYKni"
                                        >
                                          <Typography
                                            className="typography"
                                            color="tertiary"
                                            fontFamily="primary"
                                            fontStyle="normal"
                                            fontWeight="bold"
                                            letterSpacing="normal"
                                            lineHeight="normal"
                                            textDecoration="none"
                                            textTransform="none"
                                            type="paragraph1"
                                          >
                                            <Typographystyled
                                              as="p"
                                              className="typography"
                                              color="tertiary"
                                              fontFamily="primary"
                                              fontStyle="normal"
                                              fontWeight="bold"
                                              letterSpacing="normal"
                                              lineHeight="normal"
                                              textDecoration="none"
                                              textTransform="none"
                                              type="paragraph1"
                                            >
                                              <p
                                                className="Typographystyled-sc-1pbtdtz-0 kQqjtw typography"
                                                color="tertiary"
                                                fontFamily="primary"
                                                fontStyle="normal"
                                                fontWeight="bold"
                                                letterSpacing="normal"
                                                textDecoration="none"
                                                type="paragraph1"
                                              >
                                                Volver
                                              </p>
                                            </Typographystyled>
                                          </Typography>
                                        </div>
                                      </Fakebuttonstyled>
                                    </FakeButton>
                                  </a>
                                </div>
                              </Component>
                            </InternalLink>
                          </div>
                        </ProductDetailstyled>
                      </div>
                    </Colstyled>
                  </Col>
                </div>
              </Rowstyled>
            </Row>
          </div>
        </Containerstyled>
      </Container>
    </ProductDetail>
  </Component>
</WrapperComponent>
`;
